src/mypy_zope/plugin.py

```diff
@@ -698,7 +698,9 @@ def _apply_interface(self, impl: TypeInfo, iface: TypeInfo) -> None:
         if not any(promote in ti._promote for ti in impl.mro):
             faketi = TypeInfo(SymbolTable(), iface.defn, iface.module_name)
             faketi._promote = [promote]
-            impl.mro.append(faketi)
+            faketi.metaclass_type = iface.metaclass_type
+            # Insert the TypeInfo before the builtins.object that's at the end.
+            impl.mro.insert(len(impl.mro) - 1, faketi)
```

Comment:
Are you sure `builtins.object` is the last item each and every time? Would it make sense to add some kind of assertion on that in the code?

Reply:
I'm pretty sure since mypy will not let you put a base class of object first:
See: https://mypy-play.net/?mypy=0.981&python=3.11&gist=2ba86753872fd1a1fd4fffad9c4ded9a

I'm happy to add an assert too.

Reply:
Okay, if you can add the assert, I think we can merge this. Thanks for digging into it!